crates/database/src/lib.rs

```diff
@@ -21,33 +29,113 @@ pub enum RepositoryError {
     TargetNotFound,
 }
 
-/// Definition of a trait for repository operations
+/// Definition of a trait for repository operations.
 #[async_trait]
 pub trait Repository<Entity>: Sync + Send
 where
-    Entity: Sized + Serialize,
+    Entity: Identifiable + Sized + Serialize + Clone + Send + Sync + for<'de> Deserialize<'de> + Unpin + 'static,
```

Comment:
you can split this for better readability.
```suggestion
    Entity: Sized + Clone + Send + Sync + 'static,
    Entity: Identifiable + Unpin,
    Entity: Serialize + for<'de> Deserialize<'de>,
```
Please don't forget to autoformat the file.